pkg/apis/agones/v1/gameserver.go

```diff
@@ -572,22 +584,13 @@ func (gs *GameServer) FindGameServerContainer() (int, corev1.Container, error) {
 // ApplyToPodContainer applies func(v1.Container) to the specified container in the pod.
 // Returns an error if the container is not found.
 func (gs *GameServer) ApplyToPodContainer(pod *corev1.Pod, containerName string, f func(corev1.Container) corev1.Container) error {
-	var container corev1.Container
```

Comment:
Just decided to make this code a bit easier. In the previous implementation, containerIndex was the **last** match of a containereNames. But in general, having 2 containers with the same name is not valid, so it's ok to return the first occurrence.

Reply:
There is a `Pod` validation which prohibit having two containers in one `Pod` with the same name.